docs/semgrep-supply-chain/license-scanning.md

```diff
@@ -0,0 +1,115 @@
+---
+slug: license-scanning
+append_help_link: true
+title: License scanning 
+hide_title: true
+description: "Semgrep Supply Chain can detect and list a package's license. Prevent or exempt certain packages from being used based on its license."
+tags:
+  - Team & Enterprise tier
+  - Semgrep Supply Chain
+---
+
+# License scanning
+
+Semgrep Supply Chain's **license scanning** feature enables you to explicitly allow or disallow (block) a package's use in your repository based on its license. For example, your company policy may disallow the use of packages with the Creative Commons Attribution-NonCommercial (CC-BY-NC) license.
+
+![Screenshot of Semgrep Supply Chain Dependencies tab with licenses listed](/img/sc-license-scanning.png)
+*Figure 1*. Screenshot of Semgrep Supply Chain Dependencies tab with licenses listed.
+
+:::info Prerequisites
+* License scanning can accessed through Semgrep Cloud Platform (SCP). [Create an account](/semgrep-code/getting-started/#signing-in-to-semgrep-cloud-platform) to view and manage license workflows.
+* To detect licenses, you must [add or onboard a project](/semgrep-code/getting-started/#option-b-adding-a-repository-from-github-gitlab-or-bitbucket) (repository) to Semgrep Cloud Platform for scanning.
+:::
+
+## Viewing licenses
+
+To view a package's license:
+
+1. [Sign in to Semgrep Cloud Platform](https://semgrep.dev/login).
+2. Click **[Supply Chain](https://semgrep.dev/orgs/-/supply-chain)** > **Dependencies**. By default licenses are listed in the row of their respective package.
+3. If you don't see any licenses:
+    1. Click the Supply Chain **Settings** on the header menu. These settings are specific to Semgrep Supply Chain.
+    ![Screenshot of Semgrep Supply Chain Settings tab](/img/sc-settings.png) *Figure 2*. Screenshot of Semgrep Supply Chain Settings tab.
+    2. Click <i class="fa-solid fa-toggle-large-on"></i> **Dependency search** if it is not already enabled.
+    3. Click <i class="fa-solid fa-toggle-large-on"></i> **Show licenses** if it is not already enabled.
+4. Optional: Search for a specific license by entering its name in the search box.
+
+## Blocking, commenting, or allowing licenses 
+
+This section provides guides on blocking or allowing packages in CI pipelines based on the license of a package.
+
+### Types of license permissions
+
+Licenses in Semgrep are assigned the following types of permissions:
+
+<dl>
+<dt>Allow</dt>
+<dd>Packages with licenses assigned this type of permission are allowed for use in the codebase.</dd>
+<dt>Comment</dt>
```

Comment:
@andy-r2c Why do we have "comment" on licenses? Is it more like 'we'll let you have this, but have you considered looking for another library' or 'we'll let this pass, but only now and soon we'll have to forbid it'? In Semgrep code it is clearer to me that we have a comment mechanism to balance out the chance that something is a false positive while not speedbumping devs. Just curious if I can provide a better case for Comment for our users. Thank you!

Reply:
Similar reason -- some security teams might want to caution developers instead of full-on block and speedbump devs. One use case I've heard is it's okay to use X license if the repo is internal, but not okay if it's an external one.